habitat-baselines/habitat_baselines/rl/ddppo/algo/ddppo.py

```diff
@@ -23,16 +23,30 @@ def _recursive_apply(inp, fn):
         return fn(inp)
 
 
+def _convert_to_numpy_safe(t):
+    if t is None:
+        return None
+    if t.device.type == "cpu":
+        return t.numpy()
+    return t
+
+
 def _cpu_to_numpy(inp):
-    return _recursive_apply(
-        inp, lambda t: t.numpy() if t.device.type == "cpu" else t
-    )
+    return _recursive_apply(inp, _convert_to_numpy_safe)
+
+
+def _numpy_to_cpu_safe(t):
+    if t is None:
+        return None
+    if isinstance(t, np.ndarray):
+        return torch.from_numpy(t)
```

Comment:
I think if the default device has changed, this new tensor might not be on the CPU. You will need to explicitly put it there.

Reply:
Addressed with docstring and typing.